thetis/callback.py

```diff
@@ -271,8 +271,10 @@ def evaluate(self, index=None):
         """
         Evaluates callback and pushes values to log and hdf file (if enabled)
         """
-        values = self.__call__()
         time = self.solver_obj.simulation_time
+        if time < self.start_time or time > self.end_time:
+            return
```

Comment:
`self.start_time` and `self.end_time` are not defined in this parent class?

Reply:
Ah yes, I must have dropped those changes as well. Will look into it in more detail now.